{"cells":[{"metadata":{},"cell_type":"markdown","source":" # Deconstrunting CORD-19\n\n>  On March 13th, the White House, along the Chan-Zuckerberg Foundation and the Allen Center for AI launched the COVID-19 Open Research Dataset Challenge on Kaggle with the objective of better understanding the novel Corona Virus. For this purpose, a massive corpus of more than 45,000 scholarly articles was made available as the competition dataset, called the COVID-19 Open Research Dataset (CORD-19).\n\n> I found this initiative fascinating, and hints me of a world where science is a truly open, collaborative pursuit. For that reason, I decided to understand the story of the dataset better by asking:\n> * What can the metadata tell me about the views being represented\n> [](http://)* What are the assumptions made in a corpus of a modern, western view of science and knowledge"},{"metadata":{},"cell_type":"markdown","source":"> ## The Dataset Structure\n> The CORD-19 adds up to a total of 4GB, which can be downloaded. However, the best way to work with it is using a Kaggle notebook that can access the dataset in its workspace. Upon of the local filesystem, we see that the dataset is made up of json files, organized within direcotories."},{"metadata":{"_uuid":"8f2839f25d086af736a60e9eeb907d3b93b6e0e5","_cell_guid":"b1076dfc-b9ad-4769-8c92-a6c4dae69d19","trusted":true},"cell_type":"code","source":"import numpy as np # linear algebra\nimport pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\nimport json\nimport os\ndirectories = set()\nprints = 0\nfor dirname, _, filenames in os.walk('/kaggle/input'): \n    for filename in filenames:   \n        if prints < 10:\n            print(os.path.join(dirname, filename))\n            prints += 1","execution_count":44,"outputs":[{"output_type":"stream","text":"/kaggle/input/CORD-19-research-challenge/metadata.csv\n/kaggle/input/CORD-19-research-challenge/json_schema.txt\n/kaggle/input/CORD-19-research-challenge/metadata.readme\n/kaggle/input/CORD-19-research-challenge/COVID.DATA.LIC.AGMT.pdf\n/kaggle/input/CORD-19-research-challenge/noncomm_use_subset/noncomm_use_subset/d99acb4e99be7852aa61a688c9fbd38d44b5a252.json\n/kaggle/input/CORD-19-research-challenge/noncomm_use_subset/noncomm_use_subset/748d4c57fe1acc8d9d97cf574f7dea5296f9386c.json\n/kaggle/input/CORD-19-research-challenge/noncomm_use_subset/noncomm_use_subset/b891efc6e1419713b05ff7d89b26d260478c28df.json\n/kaggle/input/CORD-19-research-challenge/noncomm_use_subset/noncomm_use_subset/353852971069ad5794445e5c1ab6077ce23da75d.json\n/kaggle/input/CORD-19-research-challenge/noncomm_use_subset/noncomm_use_subset/76d2990a2663635e195b8a9818f9664872b6d3af.json\n/kaggle/input/CORD-19-research-challenge/noncomm_use_subset/noncomm_use_subset/5407b1aabede9eeed76c59a7b890be9f513712b7.json\n","name":"stdout"}]},{"metadata":{},"cell_type":"markdown","source":"> ## What can the directory structure tell us about embedded views in the corpus, about the standards and requirements that shaped it? \n> To start answering, we look at the unique directories and see that there are four. It seems that each one corresponds to a type a licence. In a Western, modern view science, licence rights are often more important than knowledge. Even though this dataset has been made open, its structure acknowledges that importance. "},{"metadata":{"trusted":true},"cell_type":"code","source":"directories = set()\nfor dirname, _, filenames in os.walk('/kaggle/input'):\n    for filename in filenames:\n        directories.add(dirname)\ndirectories","execution_count":52,"outputs":[{"output_type":"execute_result","execution_count":52,"data":{"text/plain":"{'/kaggle/input/CORD-19-research-challenge',\n '/kaggle/input/CORD-19-research-challenge/biorxiv_medrxiv/biorxiv_medrxiv',\n '/kaggle/input/CORD-19-research-challenge/comm_use_subset/comm_use_subset',\n '/kaggle/input/CORD-19-research-challenge/custom_license/custom_license',\n '/kaggle/input/CORD-19-research-challenge/noncomm_use_subset/noncomm_use_subset'}"},"metadata":{}}]},{"metadata":{},"cell_type":"markdown","source":"> ## Structure of an inidividual paper\n> Jsons are made up of nested key-value pairs. By looking at the outermost 'shell' of keys, we see the basic components of a paper. These components correspond to a standarized composition of scholarly articles, so it can tell us a story about the underlying system of knowledge creation. Here, we take a single paper and examine its outermost keys.\n"},{"metadata":{"_uuid":"d629ff2d2480ee46fbb7e2d37f6b5fab8052498a","_cell_guid":"79c7e3d0-c299-4dcb-8224-4455121ee9b0","trusted":true},"cell_type":"code","source":"with open('/kaggle/input/CORD-19-research-challenge/biorxiv_medrxiv/biorxiv_medrxiv/64b327001f8fa95b83dc23259a2ad617be12498c.json') as f:\n    data = json.load(f)\nlist(data.keys())","execution_count":53,"outputs":[{"output_type":"execute_result","execution_count":53,"data":{"text/plain":"['paper_id',\n 'metadata',\n 'abstract',\n 'body_text',\n 'bib_entries',\n 'ref_entries',\n 'back_matter']"},"metadata":{}}]},{"metadata":{},"cell_type":"markdown","source":"> ### Paper id\n>  In the Western view of science, each paper, each unit of knowledge contribution, must be unique, differentiable and indentifiable. In contrast, indigenous knowledge, for example, is a more amourphous fluid concept with no easily discretizable units.\n"},{"metadata":{"trusted":true},"cell_type":"code","source":"data['paper_id']","execution_count":54,"outputs":[{"output_type":"execute_result","execution_count":54,"data":{"text/plain":"'64b327001f8fa95b83dc23259a2ad617be12498c'"},"metadata":{}}]},{"metadata":{},"cell_type":"markdown","source":"> ### Metadata\n> The metadata constains the most important factor to rank the importance of the papers contribution, not by design, but by the facto. It contains the name of the researcher and its academic affiliation. In Modern Science, this has become a sort of pedigree. A paper marked with it is deemed to contain a valid contribution."},{"metadata":{"trusted":true},"cell_type":"code","source":"data['metadata']['authors']","execution_count":55,"outputs":[{"output_type":"execute_result","execution_count":55,"data":{"text/plain":"[{'first': 'Maria',\n  'middle': [],\n  'last': 'Nevot',\n  'suffix': '',\n  'affiliation': {'laboratory': '',\n   'institution': 'Universitat Autònoma de 6 Barcelona (UAB)',\n   'location': {'settlement': 'Badalona', 'country': 'Spain'}},\n  'email': ''},\n {'first': 'Ana',\n  'middle': [],\n  'last': 'Jordan-Paiz',\n  'suffix': '',\n  'affiliation': {},\n  'email': ''},\n {'first': 'Glòria',\n  'middle': [],\n  'last': 'Martrus',\n  'suffix': '',\n  'affiliation': {'laboratory': '',\n   'institution': 'Universitat Autònoma de 6 Barcelona (UAB)',\n   'location': {'settlement': 'Badalona', 'country': 'Spain'}},\n  'email': ''},\n {'first': 'Cristina',\n  'middle': [],\n  'last': 'Andrés',\n  'suffix': '',\n  'affiliation': {'laboratory': '',\n   'institution': 'Universitat Autònoma de 6 Barcelona (UAB)',\n   'location': {'settlement': 'Badalona', 'country': 'Spain'}},\n  'email': ''},\n {'first': 'Damir',\n  'middle': [],\n  'last': 'García-3 Cehic',\n  'suffix': '',\n  'affiliation': {'laboratory': 'Current address: Virology Unit, Microbiology Department, Hospital Universitari Vall',\n   'institution': 'Universitat Autònoma de Barcelona',\n   'location': {'settlement': 'Barcelona', 'country': 'Spain'}},\n  'email': ''},\n {'first': 'Josep',\n  'middle': [],\n  'last': 'Gregori',\n  'suffix': '',\n  'affiliation': {'laboratory': \"Liver Unit, Liver Disease Laboratory-Viral Hepatitis, Internal Medicine Department, 13 Vall d'Hebron Institut Recerca (VHIR)-Hospital Universitari Vall d'Hebron (HUVH)\",\n   'institution': '',\n   'location': {'postCode': '14',\n    'settlement': 'Barcelona',\n    'country': 'Spain'}},\n  'email': ''},\n {'first': 'Sandra',\n  'middle': [],\n  'last': 'Franco',\n  'suffix': '',\n  'affiliation': {'laboratory': '',\n   'institution': 'Universitat Autònoma de 6 Barcelona (UAB)',\n   'location': {'settlement': 'Badalona', 'country': 'Spain'}},\n  'email': ''},\n {'first': 'Josep',\n  'middle': [],\n  'last': 'Quer',\n  'suffix': '',\n  'affiliation': {'laboratory': \"Liver Unit, Liver Disease Laboratory-Viral Hepatitis, Internal Medicine Department, 13 Vall d'Hebron Institut Recerca (VHIR)-Hospital Universitari Vall d'Hebron (HUVH)\",\n   'institution': '',\n   'location': {'postCode': '14',\n    'settlement': 'Barcelona',\n    'country': 'Spain'}},\n  'email': ''},\n {'first': 'Miguel',\n  'middle': ['Angel'],\n  'last': 'Martinez',\n  'suffix': '',\n  'affiliation': {'laboratory': '',\n   'institution': 'Universitat Autònoma de 6 Barcelona (UAB)',\n   'location': {'settlement': 'Badalona', 'country': 'Spain'}},\n  'email': 'mmartinez@irsicaixa.es'}]"},"metadata":{}}]},{"metadata":{},"cell_type":"markdown","source":"> ### Abstract \n> Science is atomic and made up from individually, separable and fully differenciable pieces of knowledge. It follows then, that there must be an abstract that as the University of Melbourne recommends: 'must be fully self-contained and make sense by itself, without further reference to outside sources or to the actual paper.'"},{"metadata":{"trusted":true},"cell_type":"code","source":"data['abstract'][0]['text']","execution_count":56,"outputs":[{"output_type":"execute_result","execution_count":56,"data":{"text/plain":"'One unexplored aspect of HIV-1 genetic architecture is how codon choice influences 27 population diversity and evolvability. Here we compared the development of HIV-1 28 resistance to protease inhibitors (PIs) between wild-type (WT) virus and a synthetic 29 virus (MAX) carrying a codon-pair re-engineered protease sequence including 38 (13%) 30 synonymous mutations. WT and MAX viruses showed indistinguishable replication in 31 MT-4 cells or PBMCs. Both viruses were subjected to serial passages in MT-4 cells 32 with selective pressure from the PIs atazanavir (ATV) and darunavir (DRV). After 32 33 successive passages, both the WT and MAX viruses developed phenotypic resistance to 34'"},"metadata":{}}]},{"metadata":{},"cell_type":"markdown","source":"> ### Text body\n> The text body section of the json file contains text but also information on references and cites. While science is created from unique contributions, differenciable for the rest, each component is interconnected. However, this interconnections must remain fully traceable: reference and cites fulfill that purpose."},{"metadata":{"trusted":true},"cell_type":"code","source":"data['body_text']","execution_count":57,"outputs":[{"output_type":"execute_result","execution_count":57,"data":{"text/plain":"[{'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/315366 doi: bioRxiv preprint 5 mechanisms within the innate immune response (11, 29, 30) , as well as to resolve the 87 importance of codon usage in the temporal regulation of viral gene expression (31). In (Table 1) . WTp32 and MAXp32, respectively, showed 5-129 fold and 13-fold increases in IC 50 for ATV, and 6-fold and 10-fold increases in IC 50 for 130 DRV (Table 1) . Although MAXp32 displayed a higher resistance to ATV and DRV 131 than WTp32 (Table1), these differences were not significant (P = 0.4816 and P = The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/315366 doi: bioRxiv preprint compared the frequencies of resistant mutations. For each of the two studied viruses and 137 the two tested drugs, we sequenced between 1.9 × 10 7 and 4.1 × 10 7 individual protease 138 nucleotides (Table 2) . Sequence clonal analysis revealed no resistance-associated 139 substitutions in viruses propagated without drugs (Table 3) . On the other hand, both WT 140 and MAX viruses propagated in the presence of PIs developed previously described 141 resistance mutations to ATV and DRV (Table 3) . Moreover, the resistance variant 142 repertoire differed between the MAX and WT viruses (Table 3) . Specifically, the G16E (Table 2) , was also included in this analysis. We found that the five generated mutants 165 displayed comparable IC 50 values to ATV in both backgrounds, WT and MAX (Table   166 3). We also determined the replication capacity of these two ten mutant viruses in MT-4 167 cells with and without ATV (Fig. 2) . Similar to the IC50 results, the five tested variants The amino acid mutant repertoire also differed between the WT and MAX 176 viruses when they were propagated without drugs (Table 2) . Only one variant, D30N, 177 was detected in both virus populations. We do not know whether the observed variants 178 are adaptive or neutral mutations. Regardless, completely different mutant spectra were 179 detected in these two viruses. 180 We performed a maximum likelihood phylogenetic reconstruction of all WT and 181 MAX unique amino acid variants that were recovered after 32 MT-4 cell passages in the 182 presence of ATV or DRV. Remarkably, the results showed that the two viruses, which 183 shared an identical starting amino acid sequence, followed different evolutionary 184 trajectories (Fig. 3) . Upon visual inspection of these phylogenetic trees, it was also 185 . CC-BY-NC-ND 4.0 International license author/funder. It is made available under a The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/315366 doi: bioRxiv preprint 9 apparent that the MAX protease generated higher amino acid variant diversity (see 186 below). 187 We next compared the overall population nucleotide diversity of the WT and 188 MAX proteases after 32 passages in the absence or presence of ATV or DRV (Table 4) . 189 Overall, nucleotide sequence diversity was significantly higher in MAX populations such that viral proteins are translated from mRNAs using host cellular machinery. 221 We found that the WT and MAX protease viruses displayed different patterns of The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/315366 doi: bioRxiv preprint In some instances, the different mutant repertoire within the MAX protease 235 background can be easily explained by proximity within the corresponding sequence 236 space (e.g., G16E and L89I). However, with regards to other mutations, the explanation 237 for the difference is not readily apparent (e.g., S37P, G48L, Q58E and I84V). In 238 particular, the G48L substitution is very rarely selected in vivo in patients undergoing PI 239 therapy (37). However, when the above substitutions (S37P, G48L, Q58E, I84V or 240 L89I) were introduced in the WT sequence they showed parallel replication capacities 241 to those observed with a MAX background. We can speculate that the introduced 242 synonymous substitutions affected neighboring residues (e.g., RNA structure).',\n  'cite_spans': [{'start': 178, 'end': 182, 'text': '(11,', 'ref_id': None},\n   {'start': 183, 'end': 186, 'text': '29,', 'ref_id': None},\n   {'start': 187, 'end': 190, 'text': '30)', 'ref_id': None},\n   {'start': 2112, 'end': 2115, 'text': '180', 'ref_id': None},\n   {'start': 2862, 'end': 2865, 'text': '187', 'ref_id': None},\n   {'start': 3030, 'end': 3033, 'text': '189', 'ref_id': None},\n   {'start': 3199, 'end': 3202, 'text': '221', 'ref_id': None}],\n  'ref_spans': [{'start': 309,\n    'end': 318,\n    'text': '(Table 1)',\n    'ref_id': 'TABREF1'},\n   {'start': 465, 'end': 474, 'text': '(Table 1)', 'ref_id': 'TABREF1'},\n   {'start': 951, 'end': 960, 'text': '(Table 2)', 'ref_id': 'TABREF2'},\n   {'start': 1076, 'end': 1085, 'text': '(Table 3)', 'ref_id': 'TABREF3'},\n   {'start': 1240, 'end': 1249, 'text': '(Table 3)', 'ref_id': 'TABREF3'},\n   {'start': 1340, 'end': 1349, 'text': '(Table 3)', 'ref_id': 'TABREF3'},\n   {'start': 1375, 'end': 1384, 'text': '(Table 2)', 'ref_id': 'TABREF2'},\n   {'start': 1541, 'end': 1553, 'text': '(Table   166', 'ref_id': 'TABREF1'},\n   {'start': 1673, 'end': 1681, 'text': '(Fig. 2)', 'ref_id': None},\n   {'start': 1860, 'end': 1869, 'text': '(Table 2)', 'ref_id': 'TABREF2'},\n   {'start': 2472, 'end': 2480, 'text': '(Fig. 3)', 'ref_id': 'FIGREF23'},\n   {'start': 3018, 'end': 3027, 'text': '(Table 4)', 'ref_id': 'TABREF4'}],\n  'section': ''},\n {'text': 'However, it must be noted that the MAX and WT proteases have similar RNA folding 244 free energy (10). The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/315366 doi: bioRxiv preprint profile. Synonymous codons differ in their propensity to mutate and, as previously 260 suggested (25, 32, 42), this differential access to protein sequence space may affect 261 adaptive pathways.',\n  'cite_spans': [],\n  'ref_spans': [],\n  'section': '243'},\n {'text': 'Another intriguing finding of our study is that the MAX virus showed higher The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/315366 doi: bioRxiv preprint exponential growth phase using maximum likelihood methods as previously described 309 (10). The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/315366 doi: bioRxiv preprint 16 abundances below 0.5% were filtered out. All computations were performed in the R 359 language and platform, using in-house developed scripts as well as the packages 360 Biostrings (R package 2.24.1, 2012), Ape (51), and Seqinr (52).',\n  'cite_spans': [],\n  'ref_spans': [],\n  'section': '262'},\n {'text': 'Virus population genetic diversity (p-distance) was determined using the The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/315366 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/315366 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/315366 doi: bioRxiv preprint',\n  'cite_spans': [],\n  'ref_spans': [],\n  'section': '361'}]"},"metadata":{}}]},{"metadata":{},"cell_type":"markdown","source":"> ### Bibliography and references\n> This is likely the most defining component of modern science. Valid contributions are only those who build on top of other valid contributions. \n>**What happens when everything is based on something false and the mistake is never detected?**"},{"metadata":{"trusted":true},"cell_type":"code","source":"data['bib_entries']['BIBREF1']","execution_count":61,"outputs":[{"output_type":"execute_result","execution_count":61,"data":{"text/plain":"{'ref_id': 'b1',\n 'title': 'Exposing synonymous mutations',\n 'authors': [],\n 'year': None,\n 'venue': 'Trends in genetics : TIG',\n 'volume': '30',\n 'issn': '',\n 'pages': '308--321',\n 'other_ids': {}}"},"metadata":{}}]},{"metadata":{},"cell_type":"markdown","source":"> ## Retrieving the data\nWe will know retireve the data into a Python datastructure to work with it. Initially, we will put it all in a list."},{"metadata":{"trusted":true},"cell_type":"code","source":"groups = ['custom_license','noncomm_use_subset','comm_use_subset','biorxiv_medrxiv']\nall_jsons = []\nfor dirname, _, filenames in os.walk('/kaggle/input'):\n    for filename in filenames:\n        if('.json' in os.path.join(dirname, filename)):\n            with open(os.path.join(dirname, filename)) as f:\n                data = json.load(f)\n            all_jsons.append(data)","execution_count":null,"outputs":[]},{"metadata":{},"cell_type":"markdown","source":"> ## What can the metadata tell us?\n> First, we will take a look at the authors information and we start by asking how many papers don't have any authors: 9%, a fair amount. Where does this knowledge come from? \n"},{"metadata":{"trusted":true},"cell_type":"code","source":"count = 0\nfor i in all_jsons:\n    if (i['metadata']['authors'] != []):\n        count = count + 1\n\ntotal_empty = len(all_jsons) - count\nempty_proportion = total_empty/len(all_jsons)\nprint(empty_proportion*100)","execution_count":63,"outputs":[{"output_type":"stream","text":"12.135922330097088\n","name":"stdout"}]},{"metadata":{},"cell_type":"markdown","source":"> ## How many authors are there, or how many human minds have contributed to the corpus?\n> 198,297 authors are counted. "},{"metadata":{"trusted":true},"cell_type":"code","source":"authors = 0\nfor i in all_jsons:\n    if (i['metadata']['authors'] != []):\n        \n        for j in i['metadata']['authors']:\n            authors += 1\nauthors","execution_count":28,"outputs":[{"output_type":"execute_result","execution_count":28,"data":{"text/plain":"198297"},"metadata":{}}]},{"metadata":{},"cell_type":"markdown","source":"> ## Are each one of them an individual mind? Or are they minds that live in an environment, within a infrastructure, within a social setting? \n> Can examining their institutions give us a hint? First, we look at how many of them have an affiliation: 62% \n"},{"metadata":{"trusted":true},"cell_type":"code","source":"no_affil = 0\naffil = 0\nfor i in all_jsons:\n    if (i['metadata']['authors'] != []):\n        for j in i['metadata']['authors']:\n            if(j['affiliation'] == {}):\n                no_affil = no_affil + 1\n            else:\n                affil = affil + 1\n                \ntotal_authors = no_affil + affil\nno_affil_share = no_affil/total_authors\nprint((1-no_affil_share)*100)\n            ","execution_count":29,"outputs":[{"output_type":"stream","text":"62.69131656051277\n","name":"stdout"}]},{"metadata":{},"cell_type":"markdown","source":"> ## Who are those institutions?\n> We present a table with the most represented institutions in the authors affiliations:"},{"metadata":{"trusted":true},"cell_type":"code","source":"institutions_dict = {}\n\nfor i in all_jsons:\n    if (i['metadata']['authors'] != []):\n        \n        for j in i['metadata']['authors']:\n\n            if(j['affiliation'] != {}):\n                if(str(j['affiliation']['institution']) in institutions_dict.keys()):\n                    \n                    institutions_dict[str(j['affiliation']['institution'])] += 1\n                else:\n                    institutions_dict[str(j['affiliation']['institution'])] = 1\n                    \ninstitutions_dict.values()\nins_df = pd.DataFrame(\n    {\n        'Institution': list(institutions_dict.keys()),\n        'Times Appeared': list(institutions_dict.values())\n    }\n)\n\nins_df.sort_values(by='Times Appeared', ascending = False, inplace = True)\nins_df = ins_df[ins_df['Institution'] != '']\nins_df.iloc[0:20]\n","execution_count":30,"outputs":[{"output_type":"execute_result","execution_count":30,"data":{"text/plain":"                                       Institution  Times Appeared\n213                    Chinese Academy of Sciences            1890\n236                  National Institutes of Health            1297\n47                        University of California            1255\n416                    The University of Hong Kong            1011\n192       Chinese Academy of Agricultural Sciences             816\n25                                Fudan University             649\n32                                Wuhan University             554\n139                       University of Washington             541\n539            The Chinese University of Hong Kong             533\n393                             Utrecht University             528\n204               Huazhong Agricultural University             522\n31   Huazhong University of Science and Technology             453\n72       National Institute of Infectious Diseases             437\n5                            University of Toronto             432\n738                      The Ohio State University             404\n194               National University of Singapore             404\n177                         Harvard Medical School             389\n212                             Ministry of Health             381\n623                        University of Minnesota             379\n353                            Tsinghua University             378","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Institution</th>\n      <th>Times Appeared</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>213</th>\n      <td>Chinese Academy of Sciences</td>\n      <td>1890</td>\n    </tr>\n    <tr>\n      <th>236</th>\n      <td>National Institutes of Health</td>\n      <td>1297</td>\n    </tr>\n    <tr>\n      <th>47</th>\n      <td>University of California</td>\n      <td>1255</td>\n    </tr>\n    <tr>\n      <th>416</th>\n      <td>The University of Hong Kong</td>\n      <td>1011</td>\n    </tr>\n    <tr>\n      <th>192</th>\n      <td>Chinese Academy of Agricultural Sciences</td>\n      <td>816</td>\n    </tr>\n    <tr>\n      <th>25</th>\n      <td>Fudan University</td>\n      <td>649</td>\n    </tr>\n    <tr>\n      <th>32</th>\n      <td>Wuhan University</td>\n      <td>554</td>\n    </tr>\n    <tr>\n      <th>139</th>\n      <td>University of Washington</td>\n      <td>541</td>\n    </tr>\n    <tr>\n      <th>539</th>\n      <td>The Chinese University of Hong Kong</td>\n      <td>533</td>\n    </tr>\n    <tr>\n      <th>393</th>\n      <td>Utrecht University</td>\n      <td>528</td>\n    </tr>\n    <tr>\n      <th>204</th>\n      <td>Huazhong Agricultural University</td>\n      <td>522</td>\n    </tr>\n    <tr>\n      <th>31</th>\n      <td>Huazhong University of Science and Technology</td>\n      <td>453</td>\n    </tr>\n    <tr>\n      <th>72</th>\n      <td>National Institute of Infectious Diseases</td>\n      <td>437</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>University of Toronto</td>\n      <td>432</td>\n    </tr>\n    <tr>\n      <th>738</th>\n      <td>The Ohio State University</td>\n      <td>404</td>\n    </tr>\n    <tr>\n      <th>194</th>\n      <td>National University of Singapore</td>\n      <td>404</td>\n    </tr>\n    <tr>\n      <th>177</th>\n      <td>Harvard Medical School</td>\n      <td>389</td>\n    </tr>\n    <tr>\n      <th>212</th>\n      <td>Ministry of Health</td>\n      <td>381</td>\n    </tr>\n    <tr>\n      <th>623</th>\n      <td>University of Minnesota</td>\n      <td>379</td>\n    </tr>\n    <tr>\n      <th>353</th>\n      <td>Tsinghua University</td>\n      <td>378</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{},"cell_type":"markdown","source":"> ## What countries are represented by those institutions?\n> We present a table with the most frequent locations in these institutions. We observe that mostly caucasian and asian countries are represented. In the context of finding useful knowledge for fighting the pandemic, how do we know the solutions are applicable to Latin America or Africa?"},{"metadata":{"trusted":true},"cell_type":"code","source":"countries_dict = {}\nfor i in all_jsons:\n    if (i['metadata']['authors'] != []):\n        for j in i['metadata']['authors']:\n            if(j['affiliation'] != {}):\n                if ('country' in j['affiliation']['location'].keys()):\n                    if(str(j['affiliation']['location']['country']) in countries_dict.keys()):\n\n                        countries_dict[str(j['affiliation']['location']['country'])] += 1\n                    else:\n                        countries_dict[str(j['affiliation']['location']['country'])] = 1\n                    \ncoun_df = pd.DataFrame(\n    {\n        'Country': list(countries_dict.keys()),\n        'Times Appeared': list(countries_dict.values())\n    }\n)\n\ncoun_df.sort_values(by='Times Appeared', ascending = False, inplace = True)\ncoun_df = coun_df[coun_df['Country'] != '']\ncoun_df.iloc[0:10]\n\n","execution_count":31,"outputs":[{"output_type":"execute_result","execution_count":31,"data":{"text/plain":"          Country  Times Appeared\n1           China           17981\n0             USA           17652\n21          Japan            4035\n28         France            3863\n3          Canada            3730\n12             UK            3294\n4         Germany            3290\n40  United States            2749\n9           Italy            2702\n46         Taiwan            2423","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Country</th>\n      <th>Times Appeared</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>1</th>\n      <td>China</td>\n      <td>17981</td>\n    </tr>\n    <tr>\n      <th>0</th>\n      <td>USA</td>\n      <td>17652</td>\n    </tr>\n    <tr>\n      <th>21</th>\n      <td>Japan</td>\n      <td>4035</td>\n    </tr>\n    <tr>\n      <th>28</th>\n      <td>France</td>\n      <td>3863</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>Canada</td>\n      <td>3730</td>\n    </tr>\n    <tr>\n      <th>12</th>\n      <td>UK</td>\n      <td>3294</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>Germany</td>\n      <td>3290</td>\n    </tr>\n    <tr>\n      <th>40</th>\n      <td>United States</td>\n      <td>2749</td>\n    </tr>\n    <tr>\n      <th>9</th>\n      <td>Italy</td>\n      <td>2702</td>\n    </tr>\n    <tr>\n      <th>46</th>\n      <td>Taiwan</td>\n      <td>2423</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{},"cell_type":"markdown","source":" # Final thoughts: \n> This has been a first excersice in trying to understand the fascinating COVID-19 Open Research Dataset. While most Kaggle participants are diving right into the abstracts or the text in search for information, it can be useful to take a step back and look the underlying structure of the data. With that, we can ask better questions and understand the limitations of our answers. "}],"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"pygments_lexer":"ipython3","nbconvert_exporter":"python","version":"3.6.4","file_extension":".py","codemirror_mode":{"name":"ipython","version":3},"name":"python","mimetype":"text/x-python"}},"nbformat":4,"nbformat_minor":4}